### ISSUE TITLE
`ExternalProcessMemory` throws `NullReferenceException` when `createInjector = true`

### ISSUE THREAD

**Issue by psytrx** [aevitas/bluerain#7]
MCVE:

```
    internal class Program
    {
        private static void Main(string[] args)
        {
            var dllPath = @"C:\mydll.dll";
            var process = Process.GetProcessesByName("csgo").Single();
            var memory = new ExternalProcessMemory(process, true);
            var module = memory.Injector.Inject(dllPath);
        }
    }
```

The `ExternalProcessMemory` constructor throws a `NullReferenceException` when being called with `createInjector = true`. This is due to the constructor in question calling its base constructor (in which the Injector gets instantiated) before the ProcessHandle gets instantiated, which is required for injection.

See https://github.com/aevitas/bluerain/blob/v2/src/BlueRain/ExternalProcessMemory.cs#L36

My suggestion is to move
`ProcessHandle = OpenProcess(access, false, process.Id);`
up into its base constructor with sane `ProcessAccess` defaults.

Any thoughts?

**Comment by psytrx** [comment]
Another approach might be removing the `createInjector` parameter from the constructors which don't expect `ProcessAccess` as an argument in order to prevent using shady defaults.

**Comment by aevitas** [comment]
Fixed in #8 . Moving the `ProcessHandle` to `NativeMemory`, as it's internal, which allows us to simply call `OpenProcess` in the native memory ctor when we're external (local procs don't require the call anyway)
